### DISCUSSION TITLE
Data out from esp32 with SPI instead of I2S

### DISCUSSION THREAD

**Discussion by AvishkaDon** [pschatzmann/ESP32-A2DP#177] [Q&A] [answered]
Hi there,

I got a DAC (MAX541CEPA+) that only takes SPI. I was wondering how can I adapt this library to work with SPI instead of I2S.

Many thanks

**Comment by pschatzmann** [comment, marked as answer]
I suggest that you have a look at my solution which is supporting spdif: https://github.com/pschatzmann/arduino-audio-tools/tree/main/examples/examples-basic-api/basic-a2dp-spdif

Please note that you receive the data as int16_t with 2 channels, so you will need to convert it to a mono before writing it out.

I assume you know how to write the data to your DAC because I can't help you with that.
